File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
import java.util.Arrays;
import java.util.stream.Collectors;
class Solution {
    public int solution(String myString, String pat) {
        String sorted = Arrays.stream(myString.split("")).map((item) -> {
            if (item.equals("A")) return item = "B";
            if (item.equals("B")) return item = "A";
            return item;
        }).collect(Collectors.joining(""));
        return sorted.contains(pat) ? 1 : 0;
    }
}

File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
    let sorted = myString.split("").map((item) => {
      if (item === "A") return item = "B"
      if (item === "B") return item = "A"
   }).join("");
   return sorted.includes(pat) ? 1 : 0;
}